Makefile

```diff
@@ -1,107 +1,121 @@
-       Serif-Regular \
-       Serif-Semibold \
-       Serif-Bold \
-       Serif-Italic \
-       Serif-SemiboldItalic \
-       Serif-BoldItalic \
-       SerifDisplay-Regular \
-       Math-Regular \
-       SerifInitials-Regular \
-       Mono-Regular \
-       Keyboard-Regular \
-       $(NULL)
-
-SFD=$(FONTS:%=$(SOURCEDIR)/$(NAME)%.sfd)
-NRM=$(FONTS:%=$(BUILDDIR)/$(NAME)%.nrm)
-CHK=$(FONTS:%=$(BUILDDIR)/$(NAME)%.chk)
-OTF=$(FONTS:%=$(NAME)%.otf)
-SVG=preview.svg
-PDF=$(DOC)/Opentype-Features.pdf $(DOC)/Sample.pdf $(DOC)/Math-Sample.pdf
+NAME = Libertinus
+VERSION = 6.11
+
+MAKEFLAGS := -s -j$(shell nproc) -Otarget
+SHELL = bash
+
+DIST = $(NAME)-$(VERSION)
+
+SOURCEDIR = sources
+BUILDDIR = build
+GSUB = $(SOURCEDIR)/features/gsub.fea
+DOCSDIR = documentation
+TOOLSDIR = tools
+
+PY ?= python3
+BUILD = $(TOOLSDIR)/build.py
+NORMALIZE = $(TOOLSDIR)/sfdnormalize.py
+
+# Default to explicitly enumerating fonts to build;
+# use `make ALLFONTS=true ...` to build all *.sfd files in the source tree or
+# use `make FONTS="Face-Style" ...` to make targets for only particular font(s).
+ALLFONTS ?= false
+
+# Canonical list of fonts face / and style combinations to build;
+# note that order here will be used for some documentation
+SERIF_STYLES := Regular Semibold Bold Italic SemiboldItalic BoldItalic
+SANS_STYLES  := Regular Bold Italic
+REGULAR_ONLY := Math Mono Keyboard SerifDisplay SerifInitials
+
+ifeq ($(ALLFONTS),true)
+	FONTS := $(notdir $(basename $(wildcard $(SOURCEDIR)/*.sfd)))
+else
+	FONTS ?= $(foreach STYLE,$(SERIF_STYLES),$(NAME)Serif-$(STYLE)) \
+			 $(foreach STYLE,$(SANS_STYLES),$(NAME)Sans-$(STYLE)) \
+			 $(foreach FACE,$(REGULAR_ONLY),$(NAME)$(FACE)-Regular)
+endif
+
+# Generate lists of various intermediate forms
+SFD = $(addsuffix .sfd,$(addprefix $(SOURCEDIR)/,$(FONTS)))
+NRM = $(addsuffix .nrm,$(addprefix $(BUILDDIR)/,$(FONTS)))
+CHK = $(addsuffix .chk,$(addprefix $(BUILDDIR)/,$(FONTS)))
+
+# Generate list of final output forms
+OTF = $(addsuffix .otf,$(FONTS))
+SVG = preview.svg
+PDF = $(DOCSDIR)/Opentype-Features.pdf $(DOCSDIR)/Sample.pdf $(DOCSDIR)/Math-Sample.pdf
 
 export SOURCE_DATE_EPOCH ?= 0
 
 .SECONDARY:
+.ONESHELL:
 
+.PHONY: all otf doc normalize check
 all: otf $(SVG)
-
 otf: $(OTF)
 doc: $(PDF)
 normalize: $(NRM)
 check: $(CHK)
 
-
 nofea=$(strip $(foreach f,Initials Keyboard Mono,$(findstring $f,$1)))
 
-$(BUILDDIR)/%.otl.otf: $(SOURCEDIR)/%.sfd $(GSUB) $(BUILD)
-	@echo "      BUILD  $(*F)"
-	@mkdir -p $(BUILDDIR)
-	@$(PY) $(BUILD)                                                        \
-		--input=$<                                                     \
-		--output=$@                                                    \
-		--version=$(VERSION)                                           \
-		$(if $(call nofea,$@),,--feature-file=$(GSUB))                 \
-		;
+$(BUILDDIR):
+	mkdir -p $@
+
+$(BUILDDIR)/%.otl.otf: $(SOURCEDIR)/%.sfd $(GSUB) $(BUILD) | $(BUILDDIR)
+	$(info       BUILD  $(*F))
```

Comment:
The non-breaking space should be explicitly mentioned in a comment, otherwise it could easily be overlooked and inadvertently removed, or otherrwise cause confusion when searching or editing this file.

Reply:
This would need to be commented all over the place, and the kind of people that would _understand_ such comments and bother with them are people that will probably catch this anyways. If not we can catch it on PRs. It isn't going to break people's builds if they muck around with this and get it wrong, it just won't line up the pretty messages. I think the cost of fixing in confusion and clutter this is higher than the risk of what could break.

Reply:
I’m fine with using no break space, my editor is always configured to highlight these (as I often use them elsewhere).